### PR TITLE
Update calculate gas logic

### PR DESCRIPTION
**Pull request by osipov-mit**
## Changes
- Update calculation gas logic follow https://github.com/gear-tech/gear/pull/1051
- Add test to calculate gas for reply message
---
## Bump version to 0.21.0

### CODE REVIEW

**Review by gshep — APPROVED** [review]
Perfect! :gear:

File: api/src/Gas.ts [hunk 1]
```diff
@@ -91,39 +97,44 @@ export class GearGasSpent {
...
     code: Hex | Buffer,
     payload: PayloadType,
-    value: number | string,
+    value?: number | string,
+    allowOtherPanics?: boolean,
     metaOrTypeOfPayload?: string | Metadata,
-  ): Promise<u64> {
-    return await this.api.rpc['gear'].getInitGasSpent(
+  ): Promise<GasInfo> {
+    return await this.api.rpc['gear'].calculateInitGas(
```
Comment (nikitayutanov) [inline api/src/Gas.ts:132]:
```suggestion
    return this.api.rpc.gear.calculateInitGas(
```
`return await` is redudant

Comment (osipov-mit) [reply api/src/Gas.ts]: I agree about await)

File: api/src/Gas.ts [hunk 2]
```diff
@@ -187,39 +203,44 @@ export class GearGasSpent {
...
     destinationId: Hex,
     payload: PayloadType,
-    value: number | string,
+    value?: number | string,
+    allowOtherPanics?: boolean,
     metaOrTypeOfPayload?: string | Metadata,
-  ): Promise<u64> {
-    return await this.api.rpc['gear'].getHandleGasSpent(
+  ): Promise<GasInfo> {
+    return await this.api.rpc['gear'].calculateHandleGas(
```
[the remark this answers is not in the document]
Comment (osipov-mit) [reply api/src/Gas.ts]: It doesn't work

**Review by nikitayutanov — APPROVED** [review]

**Pull request merged**